### PR TITLE
Change FSE plugin GH workflow to editing toolkit

### PR DESCRIPTION
**Pull request by noahtallen**
#### Changes proposed in this Pull Request
1. Change workflow filename
2. Change workflow name
3. Change workflow comments
4. Change archive & app name (companion diff: D47220-code)

#### Testing instructions
Workflow should continue working. The sync step will start working after diff is deployed.

### CODE REVIEW

**Comment by noahtallen** [comment]
It's actually a bit different. Here's how it works:

1. The build job checks out the entire calypso repo, installs dependencies.
2. The build job then builds just the editing toolkit plugin.
3. The build job uploads _just_ the plugin as an archive. Note that `apps/full-site-editing/full-site-editing-plugin` is a proper WordPress plugin that can be installed in any WordPress instance once its code is built. So that is uploaded as an archive called "editing-toolkit-build-archive"
4. Note that the uploaded archive above is the .zip file that we can download from GH.

Then, in subsequent jobs, instead of installing dependencies and rebuilding the plugin for each step, we use the already-built archive that was uploaded in step 3. This is the code you linked to -- it extracts the archive that we had previously uploaded.

We do this because workflow jobs _do not share the same environment_. So when the "run phpunit tests" job starts, it does not have any of the environment step up which was created in the "build plugin" job.

In other words, we would have to reinstall and rebuild everything for every single job. Just downloading the artifact again is faster.

Again, this works because the plugin directory with the code built is the exact thing that gets installed on end-user sites. It's also the same thing that gets sent to create the wpcom sync diff.

Finally, this approach lets us run some of the jobs in parallel. So we run build first, and then the following jobs run in parallel after the build job finishes:
- phpunit tests
- wpcom sync diff
- newspack-blocks verification

**Comment by jeyip** [comment]
> Then, in subsequent jobs, instead of installing dependencies and rebuilding the plugin for each step, we use the already-built archive that was uploaded in step 3. This is the code you linked to -- it extracts the archive that we had previously uploaded.

I see! That makes perfect sense.

> We do this because workflow jobs do not share the same environment. So when the "run phpunit tests" job starts, it does not have any of the environment step up which was created in the "build plugin" job.

Gotcha. That seems to explain why we have all other jobs defined with the `needs: build` identifier. We want to use the previously uploaded artifact in the subsequent jobs. Thanks for clarifying Noah 👍
